### ISSUE TITLE
Validate OUI theme compliance - phase 1 

### ISSUE THREAD

**Issue by joshuarrrr** [opensearch-project/ml-commons-dashboards#217]
This is a tracking issue for phase 1 of OUI theme plugin compliance, which will involve changes to OUI typography and color SASS variables (for both dark and light mode). **For full details, including dates and tooling, see https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4291**

If your plugin has no UI views or components, please add a comment to that effect, and we'll remove the repo from future look and feel update campaigns.

## Actions required

- [x] Acknowledge the issue by adding an assignee who will serve as a point-of-contact
- [x] Provide screenshots and navigation instructions for important views and screens in your plugin or application. This will assist with both manual and automatic validation of compliance
- [ ] Identify all typographic properties and styles
- [ ] Create issues to mitigate typographic styles not inherited from OUI components or SASS variables
- [ ] Identify all color values, including visualization colors and palettes
- [ ] Create issues to mitigate typographic styles not inherited from OUI components or SASS variables
- [ ] Resolve all mitigation issues
- [ ] Validate and test plugin with updated OUI theme

## Questions?

Add a comment on https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4291

**Comment by wanglam** [comment]
Hi @joshuarrrr there are mainly two kind of screens in the admin UI now. Here are the screenshots:
- Deployed table list
![image](https://github.com/opensearch-project/ml-commons-dashboards/assets/4034161/4c5d06f7-6aa3-4313-bad4-0f4f6eb8c171)
![image](https://github.com/opensearch-project/ml-commons-dashboards/assets/4034161/b58d12ce-fd48-4307-8c7e-9f6b4b28c26d)

![image](https://github.com/opensearch-project/ml-commons-dashboards/assets/4034161/0b2acdaa-26c6-4afd-bde0-7daa38f341b8)
![image](https://github.com/opensearch-project/ml-commons-dashboards/assets/4034161/cc352d3e-0b82-419f-88a2-120b2f85bf39)

- ML node status panel
![image](https://github.com/opensearch-project/ml-commons-dashboards/assets/4034161/3f186e1f-156f-4edc-802f-60c31c0cba1b)

We will try to find out all the typographic and styles that not inherited from OUI components, and then try to mitigate them.

**Comment by joshuarrrr** [comment]
@wanglam To answer your offline question, for:

https://github.com/opensearch-project/ml-commons-dashboards/blob/e2825c48248311bf8299c9e988d2834635aadffc/public/components/preview_panel/index.tsx#L100

and 

https://github.com/opensearch-project/ml-commons-dashboards/blob/e2825c48248311bf8299c9e988d2834635aadffc/public/components/preview_panel/index.tsx#L104

The inline-styling of the font size should just be removed, and use the default sizing from `EuiDescriptionListTitle`: https://oui.opensearch.org/1.0/#/display/description-list

cc @KrooshalUX for visibility of guidance.

**Comment by joshuarrrr** [comment]
@wanglam Can you provide some instructions or pointers to sample data and models? We'd like to be able to load that into our testing endpoint and reproduce the views you've included here.

**Comment by wanglam** [comment]
Hi @joshuarrrr , Thank you for answering my questions. There is a model upload process in functional test. We can upload all the [pre-trained models](https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/#uploading-pretrained-models) by this flow.

Here is a pure API flow to deploy a model with opensearch ml-commons 2.8 installed:

- Enable ml-commons setting
```
PUT /_cluster/settings
{
  "transient": {
    "plugins.ml_commons.only_run_on_ml_node": false,
    "plugins.ml_commons.native_memory_threshold": 100
  }
}
```
- Register model group
```
POST /_plugins/_ml/model_groups/_register
{
  "name":"test-model-group",
  "model_access_mode":"public"
}
```
Response example
```
{
  "model_group_id": "jL7Z24gBBvRGm7DQ4Ip1",
  "status": "CREATED"
}
```
- Upload pre-trained model with model group id
```
POST /_plugins/_ml/models/_upload
{
  "name": "huggingface/sentence-transformers/all-MiniLM-L12-v2",
  "version": "1.0.1",
  "model_group_id": "jL7Z24gBBvRGm7DQ4Ip1",
  "model_format": "TORCH_SCRIPT"
}
```
Response example
```
{
  "task_id": "n77b24gBBvRGm7DQMIrf",
  "status": "CREATED"
}
```
- Get model id by task id
```
GET /_plugins/_ml/tasks/n77b24gBBvRGm7DQMIrf
```
Response example
```
{
  "model_id": "l_Hb24gBXxK7xf32MUso",
  "task_type": "DEPLOY_MODEL",
  "function_name": "TEXT_EMBEDDING",
  "state": "COMPLETED",
  "worker_node": [
    "znWMK6yUT7mlFUTVP5SOLQ"
  ],
  "create_time": 1687315755231,
  "last_update_time": 1687315763626,
  "is_async": true
}
```
The model id won't be exists if task state not be `COMPLETED`, maybe we need to call get task API for multi times.
- Deploy model with model id
```
POST /_plugins/_ml/models/l_Hb24gBXxK7xf32MUso/_deploy
```
Response example
```
{
  "task_id": "w77d24gBBvRGm7DQ94pR",
  "status": "CREATED"
}
```
We can use the get task API to track the deploy process, the model will be list in admin UI once deploy success.

**Comment by wanglam** [comment]
@joshuarrrr There are another customize styles in the repo. Here are the codes:
- customize [fontWeight](https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/monitoring/model_deployment_table.tsx#L101)
```
<span style={{ fontWeight: 600 }}>Not responding</span> on {planningNodesCount} of{' '}
```
- customized [backgroundColor](https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/monitoring/index.tsx#L81)
```
<div style={{ backgroundColor: darkMode ? undefined : '#fff' }}>
```
There are some style properties includes `padding`, `width` and `maxWidth`. Do we need to mitigate these?
There are some style value in the [scss file](https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/preview_panel/nodes_table.scss#L4), is that means we need to change to oui variables?

**Comment by joshuarrrr** [comment]
@AMoo-Miki When you get a chance, can you confirm the pre-trained model instructions  in https://github.com/opensearch-project/ml-commons-dashboards/issues/217#issuecomment-1599996062 are sufficient

**Comment by joshuarrrr** [comment]
> customize [fontWeight](https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/monitoring/model_deployment_table.tsx#L101)
> 
> <span style={{ fontWeight: 600 }}>Not responding</span> on {planningNodesCount} of{' '}

In general, we shouldn't be manually specifying font-weight. `600` corresponds to the `$ouiFontWeightSemiBold` SASS variable: https://github.com/opensearch-project/oui/blob/a0e0bb5dc3f6787d4f2872da72c57a80a7cafd6c/src/global_styling/variables/_typography.scss#L67

In this case, I'd recommend replacing the `<span>` tag with a vanilla `<strong>` tag.

> customized [backgroundColor](https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/monitoring/index.tsx#L81)
> 
> <div style={{ backgroundColor: darkMode ? undefined : '#fff' }}>

This should be replaced with an OUI component that sets the correct background automatically. I can review with @KrooshalUX and provide specific guidance.

**Comment by KrooshalUX** [comment]
It would be helpful to understand why the optimizations were being made - in some cases it may be because the component was detached in Figma therefore exposing exact values instead of the OUI Sass values. In some cases it may be because the designs that were handed off had minor deviations from standards, but even in this case, the standards should always be used from OUI. I believe that @lauralexis  had delivered some of these mockups prior to our internal Figma library launch, which may have lead engineering to believe optimizations should have been made.

@wanglam if you can help outline that, @lauralexis and I can provide further guidance.

**Comment by wanglam** [comment]
@KrooshalUX @lauralexis The duration picker isn't an OUI component. The background of the duration picker is `white` just like below image.
![image](https://github.com/opensearch-project/ml-commons-dashboards/assets/4034161/bf5f5429-3d0a-4ed1-b48e-d81affe65b47)
Since the input is read-only, the default background of FieldText is `rgba(211, 218, 230, 0.05)`, the UI will like below image.
![image](https://github.com/opensearch-project/ml-commons-dashboards/assets/4034161/703c005f-5e9b-4224-9ea9-2b24c85fe66d)
Could we just remove the white background and use the default FieldText readonly background?

**Comment by KrooshalUX** [comment]
@wanglam This is in regards to the auto refresh feature correct? I think I recall trying to work through this with @lauralexis during the design phase.

I am aware that the current ReadOnly state for our form controls is lacking in contrast between the active state of the form control, and perhaps that is why this customization was made to [SuperDatePicker](https://oui.opensearch.org/1.2/#/forms/super-date-picker). I am wondering if instead we need to have the SuperDatePicker functionality next to an on/off switch instead?

**Comment by wanglam** [comment]
@KrooshalUX The UI of `SuperDatePicker` isn't 100% the same as our component. There is a min refresh rate (10 seconds) in our plugin. We will show a error tip if user input a value less than 10 seconds. 
![image](https://github.com/opensearch-project/ml-commons-dashboards/assets/4034161/70aaf290-8542-41f5-a66e-243be82efcf1)
The start button will be disabled and the tip will turn red after less than 10 seconds. Seems we can't do these by using `SuperDatePicker`. Could you give me more suggestions about that?

**Comment by wanglam** [comment]
@joshuarrrr We have some customized color in the scss files. Is that means we need to replace it with oui variables. 
Here are the codes:
- https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/monitoring/index.scss#L18
- https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/preview_panel/nodes_table.scss#L4

**Comment by joshuarrrr** [comment]
> @joshuarrrr We have some customized color in the scss files. Is that means we need to replace it with oui variables? Here are the codes:
> 
>     * https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/monitoring/index.scss#L18
> 
>     * https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/preview_panel/nodes_table.scss#L4

Yeah, you'll definitely want to update both of those to use the OUI SASS variables. @lauralexis can you provide the corret variable?

**Comment by lauralexis** [comment]
@joshuarrrr @wanglam 

The following two should be: $euiLinkColor

* https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/monitoring/index.scss#L18

* https://github.com/opensearch-project/ml-commons-dashboards/blob/main/public/components/preview_panel/nodes_table.scss#L4

**Comment by KrooshalUX** [comment]
@wanglam Got it, thanks for the context refresher - its been a while since I have looked at the UI. All good to keep the higher order component you have built, please ignore my question about switching to `SuperDatePicker`.

**Comment by wanglam** [comment]
@joshuarrrr I've created [this PR](https://github.com/opensearch-project/ml-commons-dashboards/pull/227) to mitigate all the typographic and color styles. Would you mind to help me review it?
